File: client/src/parent/models/AssignmentState.ts
export enum AssignmentState {
  Assigned,
  Active,
  RequestedToCheck,
  CheckConfirmed,
  CheckRefused,
  Archived,
}

export const AssignmentStateNames = [
  "zugewiesen",
  "aktiv",
  "bestätigen",
  "bestätigt",
  "abgelehnt",
  "erledigt",
];

export function isParentActionNeeded(state: AssignmentState): boolean {
  return state === AssignmentState.RequestedToCheck;
}

export function isDone(state: AssignmentState): boolean {
  return (
    state === AssignmentState.CheckConfirmed ||
    state === AssignmentState.Archived
  );
}

File: api/MarbleCollectorApi/ViewModels/AssignmentState.cs
namespace MarbleCollectorApi.ViewModels
{
    /// <summary>
    /// Describes the state of an chore assignment.
    /// </summary>
    public enum AssignmentState
    {
        /// <summary>
        /// The chore has been assigned to a user.
        /// </summary>
        Assigned,

        /// <summary>
        /// The chore is in progress/active.
        /// </summary>
        Active,

        /// <summary>
        /// The user has finished the chore and requests a review.
        /// </summary>
        RequestedToCheck,

        /// <summary>
        /// The chore has been confirmed as done.
        /// </summary>
        CheckConfirmed,

        /// <summary>
        /// The chore has not been done correctly and the check is therefore refused.
        /// </summary>
        CheckRefused,

        /// <summary>
        /// The chore assignment is archived.
        /// </summary>
        Archived
    }
}